lib/msal-browser/src/utils/PopupUtils.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Constants, Logger } from "@azure/msal-common";
+import { BrowserConfiguration } from "../config/Configuration.js";
+import {
+    BrowserAuthErrorCodes,
+    createBrowserAuthError,
+} from "../error/BrowserAuthError.js";
+
+/**
+ * Monitors a window until it loads a url with the same origin.
+ * @param popupWindow - window that is being monitored
+ * @param timeout - timeout for processing hash once popup is redirected back to application
+ */
+export async function monitorPopupForHash(
```

Comment:
nit: IMO it's preferable to keep all the popup code together in the same file

Reply:
Testing gets really complicated when trying to mock this function without a direct namespace for some reason.